File: Exercism Codes/JavaScript/space-age.js
export const age = (planet, seconds) => {
  switch(planet) {
    case 'earth':
      return Number((seconds / (1.0 * 31557600)).toFixed(2));
    case 'mercury':
      return Number((seconds / (0.2408467 * 31557600)).toFixed(2));
    case 'venus':
      return Number((seconds / (0.61519726 * 31557600)).toFixed(2));
    case 'mars':
      return Number((seconds / (1.8808158 * 31557600)).toFixed(2));
    case 'jupiter':
      return Number((seconds / (11.862615 * 31557600)).toFixed(2));
    case 'saturn':
      return Number((seconds / (29.447498 * 31557600)).toFixed(2));
    case 'uranus':
      return Number((seconds / (84.016846 * 31557600)).toFixed(2));
    case 'neptune':
      return Number((seconds / (164.79132 * 31557600)).toFixed(2));
    default:
      throw new ReferenceError();
  }
};

File: Exercism Codes/R/space-age.R
space_age <- function(seconds, planet) {
  planets <- c("earth" = 1.0, "mercury" = 0.2408467, "venus" = 0.61519726, "mars" = 1.8808158, "jupiter" = 11.862615, "saturn" = 29.447498, "uranus" = 84.016846, "neptune" = 164.79132)
  round(seconds / (31557600 * as.numeric(planets[planet])), digits = 2)
}
